tell_me_your_secrets/__main__.py

```diff
@@ -206,10 +216,13 @@ def write_results_to_file(self):
             write_df.to_csv(file_name)
             module_logger.info(f'Completed Writing Results to File : {self.output_path}')
 
-    def run_signatures(self, file_path, content):
+    def run_signatures(self, file_path, content) -> Tuple[Optional[str], Optional[str]]:
         for signature in self.signatures:
             match_result = signature.match(file_path, content)
-            if match_result:
+            if match_result.matched_value:
+                if match_result.matched_value in self.whitelisted_strings:
+                    module_logger.debug(f'Matched {match_result.matched_value} but skipping since it is whitelisted')
+                    return None, None
```

Comment:
Not sure if an early return is the best thing to do in case of a whitelisted string.
There might be additional credentials in the same file which could be picked up by other signatures.
```suggestion
                    continue
```

Reply:
Good catch!

Reply:
Should prob refactor this to collect results - but that's for another PR